arelle/ModelDocument.py

```diff
@@ -327,7 +327,19 @@ def load(modelXbrl, uri, base=None, referringElement=None, isEntry=False, isDisc
         elif _type == Type.INSTANCE:
             modelDocument.instanceDiscover(rootNode)
         elif _type == Type.INLINEXBRL:
-            modelDocument.inlineXbrlDiscover(rootNode)
+            try:
+                modelDocument.inlineXbrlDiscover(rootNode)
+            except RecursionError as err:
+                schemaErrorCount = sum(e == "xmlSchema:syntax" for e in modelXbrl.errors)
```

Comment:
You could use the built-in method:
```suggestion
                schemaErrorCount = modelXbrl.errors.count("xmlSchema:syntax")
```